src/types/floatob.rs

```diff
@@ -54,6 +71,13 @@ impl<'source> FromPyObject<'source> for f64 {
     // PyFloat_AsDouble returns -1.0 upon failure
     #![allow(clippy::float_cmp)]
     fn extract(obj: &'source PyAny) -> PyResult<Self> {
+        #[cfg(not(Py_LIMITED_API))]
+        unsafe {
+            if obj.is_exact_instance_of::<PyFloat>() {
```

Comment:
Is the check-then-unchecked-downcast faster than just using the checked downcast?

Reply:
I use `is_exact_instance_,of` here because I don't want to go through `isinstance` machinery for subclasses (would probably be a deoptimization).

I have thought for a while there is space for a `PyAny::downcast_exact` method. If you like, I could PR that first and then use it here?

Reply:
Sounds reasonable.

Reply:
Done 👍